cpp/tests/utilities/typed_tests.hpp

```diff
@@ -106,17 +120,36 @@ static constexpr auto all_type_ids{detail::types_to_ids<AllTypes>()};
 static constexpr auto numeric_type_ids{detail::types_to_ids<NumericTypes>()};
 
 /**---------------------------------------------------------------------------*
- * @brief `std::array` of of all non-numeric `cudf::type_id`s
+ * @brief `std::array` of of all timestamp `cudf::type_id`s
  *
  * This can be used for iterating over `type_id`s for custom testing, or used in
  * GTest value-parameterized tests.
  *---------------------------------------------------------------------------**/
-static constexpr std::array<cudf::type_id, 6> non_numeric_type_ids{
-    cudf::EMPTY,     cudf::BOOL8,    cudf::DATE32,
-    cudf::TIMESTAMP, cudf::CATEGORY, cudf::STRING};
+static constexpr std::array<cudf::type_id, 5> timestamp_type_ids{
```

Comment:
This will automatically populate the array from the types you specified in `TimestampTypes`. That way there is only a single source of truth. 
```suggestion
static constexpr std::array<cudf::type_id, 5> timestamp_type_ids{ detail::types_to_ids<TimestampTypes>() };
```